heureka/ui/src/lib/slices/createFiltersSlice.js

```diff
@@ -0,0 +1,227 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { produce } from "immer"
+
+const initialFiltersState = {
+  labels: {}, // Labels for each entity: { entityName: ["label1", "label2", ...] }
+  activeFilters: {}, // Active filters for each entity: { entityName: { label1: [value1], label2: [value2_1, value2_2], ... } }
```

Comment:
I am not a fan of deeply nested state out of previous painful experience. It makes it easier to make mistakes when updating values somewhere deep in the hierarchy. The `immer` library helps with this but if you forget to use it even once you can accidentally overwrite a huge chunk of the structure. 

I would strongly suggest reworking this (maybe not now but in the future) for the state value to be as flat as possible. I.e. rather than having one state `activeFilters` that contains the active filters for all possible entities to instead have `entity1ActiveFilters`, `entity2ActiveFilters`, etc. This will result in a bit more boilerplate but it will be much easier to understand and read and also to debug when something goes wrong.

The actions should still be doable in such a way that you don't have to copy the code.

The same goes for `filterLabelValues`, `predefinedFilters`, `activePredefinedFilters`, ...

Reply:
Sure, I restructure this soon in my next PRs.